src/Storages/MergeTree/MergeTreeParallelReplicasSelectProcessor.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <Storages/MergeTree/MergeTreeBaseSelectProcessor.h>
+
+#include <memory>
+
+namespace DB
+{
+
+class MergeTreeReadPoolParallelReplicas;
+using MergeTreeReadPoolParallelReplicasPtr = std::shared_ptr<MergeTreeReadPoolParallelReplicas>;
+
+class MergeTreeParallelReplicasSelectProcessor final : public IMergeTreeSelectAlgorithm
```

Comment:
What's the difference with `MergeTreeThreadSelectAlgorithm`? Now, the only change I see is a different pool

Reply:
You're right, there is no difference and we can unify them. Initially it was done not to mix together the old code and a new one, but I can beautify it somehow.

Reply:
Deleted it.